### PR TITLE
Add pre run for deploy-pages action to emit deployment metric

### PR DESCRIPTION
**Pull request by YiMysty**
Calling Pages telemetry API before `deploy-pages` running.
Will also adjust `deploy` job as always run.

- `ncc build` did not like multi entrypoint file :( thus added an extra build for pre.js. (also it always output as index.js, thus adjust the output folder to avoid conflicts.) 
- Add the custom error unable upload artifact since we'll always run the `deploy-pages` step now.

A running example is like this https://github.com/paper-spa/minyi/runs/4306157101?check_suite_focus=true
(^ above workflow is adjusted to delete upload-artifact action to mimic build failure. )

### CODE REVIEW

File: src/pre.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  core.info(`Sending telemetry for run id ${context.workflowRun}`)
+  await axios.post(
+    telemetryUrl,
+    {github_run_id: context.workflowRun},
+    {
+      headers: {
+        Accept: 'application/vnd.github.v3+json',
+        Authorization: `Bearer ${context.githubToken}`,
+        'Content-type': 'application/json'
+      }
```
Comment (jcudit) [inline src/pre.js:27]: `User Agent` would be neat to track things like Action version over time.  Doesn't have to be in this PR though.

File: src/pre.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+  )
+}
+
+async function main() {
+  try {
+    ensureContext()
+    await emitTelemetry()
```
Comment (jcudit) [inline src/pre.js:35]: Does it make sense to try this a few times in case we hit transient errors? 🤔 that could also hurt us if retry volume was high.

**Review by jcudit — APPROVED** [review]
Looks accurate!

File: src/pre.js (hunk 2 above)
Comment (YiMysty) [reply src/pre.js:54]: adjusted to have retry~ :)

File: src/pre.js (hunk 1 above)
Comment (YiMysty) [reply src/pre.js:42]: Will add that in a separate PR :)

**Pull request merged**